src/workerd/jsg/modules.c++

```diff
@@ -289,8 +290,14 @@ v8::Local<v8::Module> compileEsmModule(
                           resourceColumnOffset,
                           resourceIsSharedCrossOrigin, scriptId, {},
                           resourceIsOpaque, isWasm, isModule);
-  v8::ScriptCompiler::Source source(jsg::v8Str(js.v8Isolate, content), origin);
-  return jsg::check(v8::ScriptCompiler::CompileModule(js.v8Isolate, &source));
+  if (external) {
+    v8::ScriptCompiler::Source source(
+        jsg::newExternalOneByteString(js.v8Isolate, content).ToLocalChecked(), origin);
```

Comment:
Currently we compile the built-ins into the binary as single-byte but that might prove problematic long term. Specifically, if the original source files contain any non-ascii characters, they're likely to be misinterpreted here. We likely want to compile those as `kj::ArrayPtr<uint16_t>` and use two-byte externals.